### DISCUSSION TITLE
Is there any way to update the firmware without flashing?

### DISCUSSION THREAD

**Discussion by Terrevue** [forkineye/ESPixelStick#718] [Q&A]
I have all of my controllers in production so it's a pain to get them in an reflash.  However, I'd like to test out the latest update.  Is there any way to update OTA?  Thanks!

**Comment by cybercop23** [comment]
I hav enot done this myself, but there's an Upgrade firmaware under the Admin tab. You just need to have the right bin fiel for your setup.
![image](https://github.com/forkineye/ESPixelStick/assets/23619433/edb9d22e-15bb-485a-bd9d-e81830cb8fd0)

**Comment by MartinMueller2003** [comment]
You can do OTA using a EFU file. You generate an EFU using the flash tool. Then you upload the EFU via the admin page on the browser.

**Comment by MartinMueller2003** [comment]
You can use the web service to create the OTA file for you: https://espixelstickwebflasher.from-ct.com:5000/

**Reply by grammyklg** [reply]
I just tried to use the webflasher to create an OTA file.  However, when I click on the Create EFU Image button nothing happens.  There is no response or indication of any activity.

One other note is that the version choices seem to be limited to only about 6 predetermined options.

**Reply by grammyklg** [reply]
Apparently this works in Chrome but not FireFox.

**Comment by MartinMueller2003** [comment]
You can download the latest dist.zip and create an EFU file locally. I just tested the web UI and when I press the Create EFU button I get the expected popup asking me where to save the efu file. Are you saying that the menu does not pop up?

**Comment by MartinMueller2003** [comment]
Version choices are set to just the last few versions that people seem to be using, and the latest version.
